File: app/src/main/java/com/rockspin/bargainbits/data/models/cheapshark/Store.java
/*
 * Copyright (c) Rockspin 2014.
 *
 * All Rights Reserved
 */

package com.rockspin.bargainbits.data.models.cheapshark;

/**
 * Store object returned from API.
 */
public class Store {

    private String storeID; //: "2",
    private String storeName; //: "Gamer's Gate"

    public String getStoreID() {
        return storeID;
    }

    public void setStoreID(final String pStoreId) {
        this.storeID = pStoreId;
    }

    public String getStoreName() {
        return storeName;
    }

    public void setStoreName(final String pStoreName) {
        this.storeName = pStoreName;
    }
}

File: app/src/main/kotlin/com/rockspin/bargainbits/data/models/Store.kt
/*
 * Copyright (c) Rockspin 2014.
 *
 * All Rights Reserved
 */

package com.rockspin.bargainbits.data.models

import android.support.annotation.VisibleForTesting
import com.google.gson.annotations.SerializedName

/**
 * Store object returned from API.
 */
data class Store (
    @SerializedName("storeID")
    val storeId: String,
    @SerializedName("storeName")
    val storeName: String,
    @SerializedName("isActive")
    private val isActiveInt: Int,
    @SerializedName("images")
    private val images: Images
) {

    val isActive: Boolean
    get() = isActiveInt == 1

    val imageUrl: String
    get() = images.logoUrl

    @VisibleForTesting data class Images(
        @SerializedName("banner")
        val bannerUrl: String,
        @SerializedName("logo")
        val logoUrl: String,
        @SerializedName("icon")
        val iconUrl: String
    )
}
